readthedocs/builds/forms.py

```diff
@@ -33,3 +39,12 @@ def save(self, *args, **kwargs):
         obj = super(VersionForm, self).save(*args, **kwargs)
         if obj.active and not obj.built and not obj.uploaded:
             trigger_build(project=obj.project, version=obj)
+
+    def clean(self):
+        cleaned_data = super(VersionForm, self).clean()
+        if self.instance.pk is not None:  # new instance only
+            if self.instance.active is True and cleaned_data['active'] is False:
+                log.info('Removing files for version %s' % self.instance.slug)
+                clear_artifacts.delay(version_pk=self.instance.pk)
```

Comment:
Why is clearing artifacts happening in the clean method? Validating a form should not have sideeffects.